hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/ec/reconstruction/TestECReconstructionCommands.java

```diff
@@ -0,0 +1,249 @@
+        .setState(ecPipeline.getPipelineState())
+        .setNodes(ImmutableList.of(node))
+        .setReplicaIndexes(indicesForSinglePipeline)
+        .build();
+  }
+
+  @BeforeEach
+  public void connectToDatanodes() {
+    clients = new ArrayList<>(datanodeDetails.size());
+    for (int i = 0; i < datanodeDetails.size(); i++) {
+      clients.add(new XceiverClientGrpc(
+          createSingleNodePipeline(pipeline, datanodeDetails.get(i), i + 1),
+          cluster.getConf()));
+    }
+  }
+
+  @AfterEach
+  public void closeClients() {
+    if (clients == null) {
+      return;
+    }
+    for (XceiverClientSpi c : clients) {
+      c.close();
+    }
+    clients = null;
+  }
+
+  private Function<Integer, Integer> chunksInReplicaFunc(int i) {
+    if (i < EC_DATA) {
+      return (keySize) -> {
+        int dataBlocks = (keySize + EC_CHUNK_SIZE - 1) / EC_CHUNK_SIZE;
+        return (dataBlocks + EC_DATA - 1 - i) / EC_DATA;
+      };
+    } else {
+      return (keySize) -> (keySize + EC_CHUNK_SIZE * EC_DATA - 1) /
+          (EC_CHUNK_SIZE * EC_DATA);
+    }
+  }
+
+  @Test
+  public void testListBlock() throws Exception {
+    for (int i = 0; i < datanodeDetails.size(); i++) {
+      final int minKeySize = i < EC_DATA ? i * EC_CHUNK_SIZE : 0;
+      int numExpectedBlocks = (int) Arrays.stream(KEY_SIZES)
+          .filter(size -> size > minKeySize).count();
+      Function<Integer, Integer> expectedChunksFunc = chunksInReplicaFunc(i);
+      int numExpectedChunks = Arrays.stream(KEY_SIZES)
+          .map(expectedChunksFunc::apply).sum();
+      try {
+        ListBlockResponseProto response = ContainerProtocolCalls.listBlock(
+            clients.get(i), containerID, null, numExpectedBlocks + 1, null);
+        Assertions.assertEquals(numExpectedBlocks, response.getBlockDataCount(),
+            "blocks count doesn't match on DN " + i);
+        Assertions.assertEquals(numExpectedChunks,
+            response.getBlockDataList().stream()
+                .mapToInt(BlockData::getChunksCount).sum(),
+            "chunks count doesn't match on DN " + i);
+      } catch (StorageContainerException e) {
+        Assertions.assertTrue(e.getMessage().contains("does not exist"));
+        Assertions.assertEquals(0, numExpectedBlocks);
+      }
+    }
+  }
+
+  public static void startCluster(OzoneConfiguration conf) throws Exception {
+
+    // Set minimum pipeline to 1 to ensure all data is written to
+    // the same container group
+    WritableECContainerProviderConfig writableECContainerProviderConfig =
+        conf.getObject(WritableECContainerProviderConfig.class);
+    writableECContainerProviderConfig.setMinimumPipelines(1);
+    conf.setFromObject(writableECContainerProviderConfig);
+
+    cluster = MiniOzoneCluster.newBuilder(conf)
+        .setNumDatanodes(NUM_DN)
+        .setScmId(SCM_ID)
+        .setClusterId(CLUSTER_ID)
+        .build();
+    cluster.waitForClusterToBeReady();
+    om = cluster.getOzoneManager();
```

Comment:
I think we are not using all of this variable right? ex: om. If we are not using let's get ride of them, we can add when needed.

Reply:
Sure.